projects/aws/bottlerocket-bootstrap/pkg/kubeadm/controlplane_join.go

```diff
@@ -83,12 +83,19 @@ func controlPlaneJoin() error {
 		return errors.Wrapf(err, "getting kubeadm version")
 	}
 
-	if isEtcdExternal, err := isClusterWithExternalEtcd(kubeconfigPath); err != nil {
+	isEtcdExternal, err := isClusterWithExternalEtcd(kubeconfigPath)
+	if err != nil {
+		return err
+	}
+
+	// nothing to do for external etcd
+	if isEtcdExternal {
+		return nil 
```

Comment:
In case of external etcd, we should not be exiting the function. Other steps in join flow are needed.